compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -1143,6 +1143,53 @@ fn link_sanitizer_runtime(sess: &Session, linker: &mut dyn Linker, name: &str) {
     }
 }
 
+fn add_profiler_libraries(sess: &Session, crate_type: CrateType, linker: &mut dyn Linker) {
+    let needs_runtime = match crate_type {
+        CrateType::Executable | CrateType::Dylib | CrateType::Cdylib | CrateType::ProcMacro => true,
```

Comment:
I don't think a profiler runtime is needed if a dylib dependency has already provided it. Also I'm not sure how profiler runtimes work in C/C++ land, but it may be necessary to add it to static libs.

Reply:
That's a mistake, Only Executable is needed since profiler libraries are static libs. Fixed.